pennylane/estimator/resources_base.py

```diff
@@ -0,0 +1,427 @@
+    __rmatmul__ = __matmul__
+
+    @property
+    def gate_counts(self):
+        r"""Produce a dictionary which stores the gate counts
+        using the operator names as keys.
+
+        Returns:
+            dict: A dictionary with operator names (str) as keys
+                and the number of occurances in the circuit (int) as values.
+        """
+        gate_counts = defaultdict(int)
+
+        for cmp_res_op, counts in self.gate_types.items():
+            gate_counts[cmp_res_op.name] += counts
+
+        return gate_counts
+
+    def __str__(self):
+        """Generates a string representation of the Resources object."""
+
+        # --- Resources: ---
+        # Total wires: 29
+        #    algorithmic wires: 20
+        #    allocated wires: 9
+        #         clean wires: 9
+        #   	  dirty wires: 0
+        # Total gates : 96
+        #  'Toffoli': 9,
+        #  'CNOT': 60,
+        #  'Hadamard': 27
+
+        wm = self.wire_manager
+        total_wires = wm.total_wires
+        total_gates = sum(self.gate_counts.values())
+
+        total_gates_str = str(total_gates) if total_gates <= 999 else f"{Decimal(total_gates):.3E}"
+        total_wires_str = str(total_wires) if total_wires <= 9999 else f"{Decimal(total_wires):.3E}"
+
+        items = "--- Resources: ---\n"
+        items += f" Total wires: {total_wires_str}\n"
+
+        qubit_breakdown_str = f"    algorithmic wires: {wm.algo_wires}\n    allocated wires: {wm.clean_wires+wm.dirty_wires}\n\t clean wires: {wm.clean_wires}\n\t dirty wires: {wm.dirty_wires}\n"
+        items += qubit_breakdown_str
+
+        items += f" Total gates : {total_gates_str}\n  "
+
+        gate_counts = self.gate_counts
+        custom_gates = []
+        default_gates = []
+        for gate_name, count in gate_counts.items():
+            if gate_name not in DefaultGateSet:
+                custom_gates.append((gate_name, count))
+            else:
+                default_gates.append((gate_name, count))
+
+        res_order = ["Toffoli", "T", "CNOT", "X", "Y", "Z", "S", "Hadamard"]
+
+        gate_order_map = {name: i for i, name in enumerate(res_order)}
+        default_gates.sort(key=lambda x: gate_order_map.get(x[0], len(res_order)))
+
+        ordered_gates = custom_gates + default_gates
+        gate_type_str = ",\n  ".join(
+            [
+                f"'{gate_name}': {Decimal(count):.3E}" if count > 999 else f"'{gate_name}': {count}"
+                for gate_name, count in ordered_gates
+            ]
+        )
+        items += gate_type_str
+
+        return items
+
+    def __repr__(self):
+        """Compact string representation of the Resources object"""
+        return f"Resources(wire_manager={self.wire_manager}, gate_types={self.gate_types})"
+
+    def _ipython_display_(self):  # pragma: no cover
+        """Displays __str__ in ipython instead of __repr__"""
+        print(str(self))
+
+
+def add_in_series(first: Resources, other) -> Resources:  # +
```

Comment:
Please remove unnecessary comments, i.e., # + you might add them to the docstring if necessary.